### PR TITLE
API-2349/update-typescript-to-v4

### PR DESCRIPTION
**Pull request by gileswells**
### Description
Upgrades Typescript from 3.9.7 to 4.0.3 which is the latest stable release. There is 4.1 in beta currently but stuck with the stable release for all the good reasons.

### Process
- [ ] Tests added or updated for change
- [ ] Docs updated
- [ ] Accessibility concerns have been considered and addressed

### Requested feedback

### CODE REVIEW

File: src/base.accessibility.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ describe('Accessibility tests', () => {
         page.on('request', mockSwagger);
       }
 
-      await page.goto(`${puppeteerHost}${path}`, { waitUntil: 'networkidle0' });
+      await page.goto(`${puppeteerHost}${path}`, { waitUntil: 'networkidle2' });
```
Comment (gileswells) [inline src/base.accessibility.ts:13]: Consider navigation to be finished when there are no more than 2 network connections for at least 500ms. networkidle0 required 0 network connections within 500ms. On my local this change slimmed down the test by 10 seconds overall with no false failures.

Comment (gileswells) [reply src/base.accessibility.ts:13]: Previous build failed at the accessibility test with `76.987 s` duration and the current build succeeded with `77.055 s` duration with this change in place. 🤷‍♂️

Comment (rtravitz) [reply src/base.accessibility.ts:13]: Nice optimization! Maybe throw a comment above about the subtleties of `networkidle2` you mentioned here?

**Review by rtravitz — APPROVED** [review]
It passes CI and deploys working review bundles. That's good in my book. 🚀

**Pull request merged**